### ISSUE TITLE
Transfer Learning?

### ISSUE THREAD

**Issue by mattignal** [yhenon/keras-frcnn#196]
Hi there,

I have a dataset that is relatively small and different from pre-trained datasets. Fine-tuning definitely helps, but I think I may need transfer learning to get the best results.

Is there any way to implement this?

Thanks in advance!

**Comment by Abhijit-2592** [comment]
@mattignal  yes, it is possible If you are using VGG-16 as your base network then it is relatively easier. If you have your own network then that will take some time implementing coz you have to modify RPN layers, ROI layers accordingly. Check my [repo](https://github.com/Abhijit-2592/keras-frcnn) (built on a fork from this repo). Check the train_frcnn.py I have made it into a callable function. 

Follow these steps for simple transfer learning using VGG-16
1) Train VGG-16 for classification using keras and store the weights
2) Use simple parser and create the text file accordingly. I made a minor change though, the file name path should look something like this foo/train/image.jpg is taken as training data, foo/valid/image.jpg is taken as vallidation set check [simple_parser.py](https://github.com/Abhijit-2592/keras-frcnn/blob/master/keras_frcnn/simple_parser.py) for more details in my repository

NOTE: The repo was deprecated recently though coz I am working on a more generalized API with this. which I will make it available soon, Check [train_frcnn.py](https://github.com/Abhijit-2592/keras-frcnn/blob/master/train_frcnn.py) docs for more details. Optionally supports validation set and tensorboard log

**Comment by mattignal** [comment]
Thanks @Abhijit-2592. Question: Does this accomplish both fine-tuning and transfer learning? I was under the impression that inputting weights just accomplishes fine-tuning [from here](https://github.com/yhenon/keras-frcnn/issues/64#issuecomment-308756930).

**Comment by Abhijit-2592** [comment]
@mattignal  From my understanding Transfer learning is basically using the CNN trained on a dataset say "imagenet" and using it as a fixed feature extractor (Freezing all the keras layers). Fine tuning on the other hand deals with how many layers you wanna freeze. Both can be achieved by modifying the train_frcnn.py file (Freezing or unfreezing the required keras layers accordingly).
